### DISCUSSION TITLE
How to do oneshot deployment? (no restart)

### DISCUSSION THREAD

**Discussion by SanariSan** [coollabsio/coolify#3447] [Q&A] [answered]
I have a SPA React app wrapped in a Dockerfile to run the build, produce static files, and place them in a pre-mounted volume (which is also mounted to a separate backend resource to host them). The frontend and backend are in separate repositories; the only connection point is a shared volume. This setup is not a monorepo with a compose file. The problem is to perform the build inside the Docker container once and then stop it.

Obviously, a container whose purpose is to run the build command will exit when done, but Coolify tries to restart it in a loop. How do I prevent it from restarting on resource exit? Apparently, I need to pass the `--restart=no` flag, but according to the docs, I can only pass these flags:

<details>
  <summary>Flags:</summary>
  <img src="https://github.com/user-attachments/assets/f4326e4c-07b6-4df9-83e0-db7444040e65">
</details>

<details>
  <summary>Restarting:</summary>
  <img src="https://github.com/user-attachments/assets/78accae7-86f5-480d-bacf-4e996526f347">
</details>

Running `docker update --restart=no container-abc-12345` works just fine; the resource goes into the `Exited` state in the dashboard. I hope this is not the only solution (or at least tell me how to run this command in the pipeline, not manually) 🙏

*test front build repo - https://github.com/SanariSan/coolify-f

**Comment by hansipie** [comment, marked as answer]
Hello ... just sharing :)
I had the same question, but I finally succeeded in working it out by using a Docker Compose deployment. In the Compose file, you can set restart: 'no'.

**Reply by SanariSan** [reply]
I see, so the only thing I should change in my pipeline is the way I launch my builder container. That’s one way to solve it. Thanks! 👍
